plutus-tx-plugin/test/IsData/Spec.hs

```diff
@@ -0,0 +1,89 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# OPTIONS_GHC   -Wno-orphans #-}
+{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:defer-errors #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:no-context #-}
+module IsData.Spec where
+
+import           Common
+import           Lib
+import           PlcTestUtils
+import           Plugin.Data.Spec
+import           Plugin.Primitives.Spec
+
+import qualified PlutusTx.Builtins      as Builtins
+import           PlutusTx.Code
+import qualified PlutusTx.IsData        as IsData
+import           PlutusTx.Plugin
+import qualified PlutusTx.Prelude       as P
+
+import qualified PlutusCore             as PLC
+import qualified PlutusCore.MkPlc       as PLC
+import qualified UntypedPlutusCore      as UPLC
+
+import           Data.Proxy
+
+IsData.unstableMakeIsData ''MyMonoData
+IsData.unstableMakeIsData ''MyMonoRecord
+IsData.unstableMakeIsData ''MyPolyData
+
+data NestedRecord = NestedRecord { unNested :: Maybe (Integer, Integer) }
+IsData.unstableMakeIsData ''NestedRecord
+
+instance P.Eq NestedRecord where
+    {-# INLINABLE (==) #-}
+    (NestedRecord i1) == (NestedRecord i2) = i1 P.== i2
+
+data WrappedBS = WrappedBS { unWrap :: Builtins.ByteString }
+IsData.unstableMakeIsData ''WrappedBS
+
+instance P.Eq WrappedBS where
+    {-# INLINABLE (==) #-}
+    (WrappedBS i1) == (WrappedBS i2) = i1 P.== i2
+
+-- 'Z' so it sorts late and this doesn't work by accident
+data Z = Z Integer
+IsData.unstableMakeIsData ''Z
+type Syn = Z
```

Comment:
Not sure what's going on here.  Some explanation might be helpful.  I can see what it's doing, but _why_?

Reply:
This is actually just copied from the `Lift` tests. I might delete this one, it's not interesting.